custom_components/tahoma/__init__.py

```diff
@@ -84,12 +84,17 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     try:
         await client.login()
-    except TooManyRequestsException:
-        _LOGGER.error("too_many_requests")
-        return False
+        devices = await client.get_devices()
+        scenarios = await client.get_scenarios()
```

Comment:
Why create 2 variables? We only use them once.

Reply:
I want to have those two API calls in the `try, catch {}` as well, however I don't want to duplicate the exception logic 3 times. This was the only solution I could think of.

Reply:
But you don't use them in your except blocks. Or perhaps you suggest that these exceptions can happen when we retrieve the devices and scenarios?

Reply:
Yes, these exceptions can happen on retrieval of the devices and scenarios. They are needed later on in the init, however this won't be reached in case of any errors.

See https://github.com/iMicknl/ha-tahoma/issues/277.